### ISSUE TITLE
AMP seq identified as non-amp

### ISSUE THREAD

**Issue by Darcy220606** [bcgsc/AMPlify#3]
Hi there,

First, the code for the app is written very well and the idea behind it is very promising !
My issue is, I have a set of test data with sequences known to be AMPs according to other (DNN based and non DNN -based network) tools. However, when i ran amplify, the sequence was identified as non-AMPs and the probabilities were very low in comparison. My protein is a posttranslational modified - mature - protein, and as far as i understood that the models uploaded on github were also trained using mature sequences from the APD db? I checked the code, to find whether there is a parameter i should tweak to specify this, but i could not find any. Do you have a suggestion why that might be the case?  There were no errors in the run besides 

CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
[time]: I tensorflow/core/common_runtime/process_util.cc:69] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.

Many thanks and looking forward for it to work .

Best,
Anan

**Comment by mark0428** [comment]
Hi @Darcy220606,

Thanks for contacting us for your issue. Yes, the model was trained with mature sequences and part of them are from APD3. Were the sequences in your test set validated in wet lab (real AMPs)? Or they are just "putative AMPs" predicted in silico from other tools?

Cheers,
Chenkai

**Comment by Darcy220606** [comment]
Hi @mark0428 

Thanks for your reply. Yes, some sequences were predicted in silico from other tools as potentially putative AMPs, and one of them validated in the lab. 

Best,
Anan

**Comment by mark0428** [comment]
Hi @Darcy220606 

Since the negative samples we used for training the model is very different from other tools (we applied more stringent criteria to select non-AMP sequences), so it is very likely that there will be differences in predicting real non-AMP sequences when running different tools. Table 1 in our manuscript shows that other tools predict a lot of non-AMP sequences to be AMPs, while ours not. As a result, it is very likely that most of the sequences in your test data are not AMPs. The validated one should just be a false negative for our model. Hope this helps!

Best,
Chenkai
